### ISSUE TITLE
Server Whitelisting Issue

### ISSUE THREAD

**Issue by makwad** [KeshiaRose/Basic-CSV-WDC#9]
Hello, can you advise on the whitelisting server settings please.  I've tried 

tsm data-access web-data-connectors add --name "WDC URL CSV" --url https://basic-csv-wdc.herokuapp.com:443 

-- and I get the following error when trying to refresh an extract:
_Unknown Failure ( status code = 5020, WebDataRequestor::InitAPIObj Unable to load URL: https://basic-csv-wdc.herokuapp.com:443/ )_

I am using it with the following CSV api (which works fine on desktop):
https://api.coronavirus.data.gov.uk/v2/data?areaType=utla&areaCode=E08000001&metric=newLFDTests&metric=uniqueCasePositivityBySpecimenDateRollingSum&metric=uniquePeopleTestedBySpecimenDateRollingSum&metric=newCasesBySpecimenDate&format=csv

**Comment by KeshiaRose** [comment]
Hi @makwad, I would suggest hosting your own version as I can update this at any time but if you choose to go with my version what you've written is exactly how you should safelist it. (I've also added this to the readme.) This error does not look like it has anything to do with the safelist, as an incorrect safelist error would look like this: 
```
An error occurred when refreshing extracts for the Datasource 'basic-csv-wdc.herokuapp.com'; Web Data Connector with URL https://basic-csv-wdc.herokuapp.com:443 is not in the list of allowed web data connectors or does not have the correct configuration. Refresh skipped. 
```
I tested out the WDC with the URL you mentioned and was able to get it to refresh successfully on Tableau Server. What version of Tableau Server are you using? Also, are there any firewalls on your server that would prevent accessing the WDC? Are you able to reach the WDC from the browser on the Server machine?

-Keshia

**Comment by makwad** [comment]
[tabprotosrv_backgrounder_1-0_2021_04_27_13_13_41.txt](https://github.com/KeshiaRose/Basic-CSV-WDC/files/6384921/tabprotosrv_backgrounder_1-0_2021_04_27_13_13_41.txt)
Hello again. I have had a good look and can't see the issue. I have attached the log to see if you can see anything.  All I can see is the following
ID	Time	Elapsed	Key	ART	Value
209	13:13:41.417	-	msg	-	Found no WDC whitelist entry in config for URL 'https://basic-csv-wdc.herokuapp.com:443/'
218	13:13:48.263	-	web-data-connector	-	Web view loading finished. Loading was not successful. Loaded url is   'https://basic-csv-wdc.herokuapp.com/'

It works OK in the browser on the Server 2020.4.0. 

My latest attempt:
tsm data-access web-data-connectors add --name "WDC URL CSV" --url https://basic-csv-wdc.herokuapp.com:443/ --secondary https://basic-csv-wdc.herokuapp.com/.*

BTW, I have another web connector which connects/works fine.  SO I don't think it's a firewall issue:
_tsm data-access web-data-connectors add --name "UK Coronavirus Data API" --url https://dataanalyticsscc.github.io:443/index.html --secondary https://dataanalyticsscc.github.io:443/.*,https://maxcdn.bootstrapcdn.com/.*,https://connectors.tableau.com/.*,https://ajax.googleapis.com/.*,https://dataanalyticsscc.github.io/.*,https://api.coronavirus.data.gov.uk/.*_

**Comment by KeshiaRose** [comment]
Aha ok, so it is a safe list issue. Don't add a secondary list though as that will block anything that is not in that list from loading, like the CSV itself. By not including the secondary list at all you allow anything to load, this means other users can use the same WDC with different CSVs without making changes to the safe list. Also, perhaps get rid of the last slash? Here is the command I use to add this WDC, also don't forget to apply the changes:
```
tsm data-access web-data-connectors add --name "CSV WDC" --url https://basic-csv-wdc.herokuapp.com:443
tsm pending-changes apply
```

**Comment by KeshiaRose** [comment]
Also, what url are you using when you connect to the WDC in Desktop? Should just be https://basic-csv-wdc.herokuapp.com without the port.

**Comment by makwad** [comment]
Hi, I have reverted back to 
_tsm data-access web-data-connectors add --name "CSV WDC" --url https://basic-csv-wdc.herokuapp.com:443

However, the URL I am using to connect to the WDC in Desktop seems to be the issue as when I put in https://basic-csv-wdc.herokuapp.com it adds **:443** on the end. I have unsuccessfully tried to amend the workbook xml which has **_path='/' port='443'_**.

So when it publishes it:
![image](https://user-images.githubusercontent.com/32903323/116403520-8e531d00-a825-11eb-914f-8082b2227532.png)
![image](https://user-images.githubusercontent.com/32903323/116402912-dcb3ec00-a824-11eb-93b2-5abfe2655541.png)

I have attached my workbook connector to see if you can see the issue or could you send me your desktop version that workd for you when you published to the server. 
[CSV URL Connector.zip](https://github.com/KeshiaRose/Basic-CSV-WDC/files/6391753/CSV.URL.Connector.zip)

Much appreciated for your help, Dilesh.

**Comment by KeshiaRose** [comment]
Hmm, I'm not sure what is going on then, I just tried the data source you sent over on a test server where I enabled the wdc using the exact snippet I sent you and I was able to refresh it: 
![image](https://user-images.githubusercontent.com/19785852/116433883-c5fa9e80-a7fe-11eb-96a7-02172879950e.png)
What version of Server are you using? Also perhaps there are some conflicting safe list settings since you've tried a few things? Do you see multiple entries for this URL when you run
```
tsm data-access web-data-connectors list
```
![image](https://user-images.githubusercontent.com/19785852/116434135-fe01e180-a7fe-11eb-8b9e-52ae08abe054.png)

**Comment by makwad** [comment]
![Capture](https://user-images.githubusercontent.com/32903323/116564259-b497bc00-a8fc-11eb-9cc5-2e2bcd0f2493.PNG)
This is what I have.  Does the data source I sent you look the same as mine.
![Capture2](https://user-images.githubusercontent.com/32903323/116564532-f9bbee00-a8fc-11eb-98bf-e3ba865c5164.PNG)
Where desktop adds  **:443** even though I just input https://basic-csv-wdc.herokuapp.com on the WDC setup page?

**Comment by KeshiaRose** [comment]
Yes, that should be fine, it adds the :443 to mine as well. I don't know what else it could be, also what version of Server are you on?

**Comment by makwad** [comment]
Server 2020.4.0
![log](https://user-images.githubusercontent.com/32903323/116572899-40611680-a904-11eb-9a39-24418f1dff7a.PNG)

**Comment by KeshiaRose** [comment]
Ok just spun up a 2020.4.0 server and ran the same data source you sent me. My logs are exactly the same as yours up until "Resource Manager: Memory info: ###### bytes" then after that it tries to check jquery against the safe list. Since you aren't seeing any safe list checks for jquery this makes me think that the server isn't even hitting the index.html page to see that it needs to check all the linked libraries. Are you sure that you are able to access the URL from that server machine? I don't have any other options to try at this point. If you can confirm that you have access to that URL from the machine then I would suggest opening a [support ticket](https://www.tableau.com/support/case).
![image](https://user-images.githubusercontent.com/19785852/116611232-fa4b8900-a8ea-11eb-86de-61eca0e9cf4b.png)

**Comment by makwad** [comment]
Aha! It looks like access to https://github.com is blocked. The other links in index.html are fine.
If I can't get github unblocked, I guess I will have to get it hosted locally (?) 
Again thanks for your help

**Comment by KeshiaRose** [comment]
Aha! Yes, I always suggest hosting your own version. Either locally or on something like Glitch or Heroku. You can easily set up a Glitch or Heroku project with this repo by using the deploy links in the [readme](https://github.com/KeshiaRose/Basic-CSV-WDC#deploy-your-own). From there you can remove the references to github.com
